SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ISwrveConversationsSDK.java

```diff
@@ -0,0 +1,33 @@
+package com.swrve.sdk.conversations;
+
+import com.swrve.sdk.conversations.engine.model.UserInputResult;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Used internally to interface the SwrveSDK (native and Unity) with conversations
+ */
+public interface ISwrveConversationsSDK {
+    File getCacheDir();
+
+    void conversationEventsCommitedByUser(SwrveCommonConversation swrveConversation, ArrayList<UserInputResult> userInputEvents);
+
+    void conversationPageWasViewedByUser(SwrveCommonConversation swrveConversation, String pageTag);
+
+    void conversationWasStartedByUser(SwrveCommonConversation swrveConversation, String startPageTag);
+
+    void conversationWasFinishedByUser(SwrveCommonConversation swrveConversation, String endPageTag, String endControlTag);
+
+    void conversationWasCancelledByUser(SwrveCommonConversation swrveConversation, String currentPageTag);
+
+    void conversationEncounteredError(SwrveCommonConversation swrveConversation, String currentPageTag, Exception e);
+
+    void conversationTransitionedToOtherPage(SwrveCommonConversation swrveConversation, String currentPageTag, String targetPageTag, String controlTag);
+
+    void conversationLinkVisitActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
+
+    void conversationDeeplinkActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
+
+    void conversationCallActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
+}
```

Comment:
Yeah, I did different stuff with this for the Unity implementation. I'll
bring that in on this side too.
On 3 Mar 2016 12:50 p.m., "dominicmarmionswrve" <notifications@github.com>
wrote:

> In
> SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ISwrveConversationsSDK.java
> <https://github.com/Swrve/swrve-android-sdk/pull/129#discussion_r54874958>
> :
>
> > +    void conversationWasStartedByUser(SwrveCommonConversation swrveConversation, String startPageTag);
> > +
> > +    void conversationWasFinishedByUser(SwrveCommonConversation swrveConversation, String endPageTag, String endControlTag);
> > +
> > +    void conversationWasCancelledByUser(SwrveCommonConversation swrveConversation, String currentPageTag);
> > +
> > +    void conversationEncounteredError(SwrveCommonConversation swrveConversation, String currentPageTag, Exception e);
> > +
> > +    void conversationTransitionedToOtherPage(SwrveCommonConversation swrveConversation, String currentPageTag, String targetPageTag, String controlTag);
> > +
> > +    void conversationLinkVisitActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
> > +
> > +    void conversationDeeplinkActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
> > +
> > +    void conversationCallActionCalledByUser(SwrveCommonConversation swrveConversation, String currentPageTag, String tag);
> > +}
>
> The design of this whole class is yuck. I know its not your design but its
> greatly highlighted now.
>
> Is the implementation need to be different per platform?
>
> Can you create another jira to tidy this up, which we'll do as part of
> this release also?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Swrve/swrve-android-sdk/pull/129/files#r54874958>.
>